src/meta-srv/src/procedure/region_migration/update_metadata.rs

```diff
@@ -46,7 +50,17 @@ impl State for UpdateMetadata {
             UpdateMetadata::Upgrade => {
                 self.upgrade_candidate_region(ctx).await?;
 
-                // TODO(weny): invalidate fe cache.
+                if let Err(err) = ctx.invalidate_table_cache().await {
+                    warn!("Failed to broadcast the invalidate table cache message during the upgrade candidate, error: {err:?}");
+                };
+                Ok(Box::new(RegionMigrationEnd))
+            }
+            UpdateMetadata::Rollback => {
+                self.rollback_downgraded_region(ctx).await?;
+
+                if let Err(err) = ctx.invalidate_table_cache().await {
+                    warn!("Failed to broadcast the invalidate table cache message during the rollback, error: {err:?}");
+                };
                 Ok(Box::new(RegionMigrationEnd))
```

Comment:
If rollback, do we still consider the migration is ended? Normally if the migration procedure is finished, caller tends to think the region is moved successfully. However, with rolling back, there's surprising.

Reply:
I added the migration abort step, and it throws a non-retryable error.